devito/cgen_utils.py

```diff
@@ -66,6 +67,7 @@ class CodePrinter(C99CodePrinter):
     :param settings: A dictionary containing relevant settings
     """
     def __init__(self, settings={}):
+        self.dtype = settings.pop('dtype', np.float32)
```

Comment:
get, not pop !

Reply:
I don't think `dtype` is a setting for `C99CodePrinter`